src/interfaces/chain.h

```diff
@@ -129,6 +130,12 @@ class Chain
     //! unlocked when the returned interface is freed.
     virtual std::unique_ptr<Lock> lock(bool try_lock = false) = 0;
 
+    //! Returns a pair:
+    //!  * first : whether the block filter for this block could be found.
+    //!  * second: whether any of the elements match the block via a BIP 157
+    //!            block filter.
+    virtual std::pair<bool, bool> filterMatchesAny(const uint256& hash, const GCSFilter::ElementSet& filter_set) = 0;
```

Comment:
This is quite a complicated interface which asks a lot from the caller. This is an indication that the calling code should be refactored to use better abstractions.

For instance, a `BlockScanner` abstraction could be used to return blocks and whose implementation may vary depending on whether block filters exist. The logic around filters could be encapsulated there rather than having the details exposed in the wallet. I think this would benefit long-term code health.